core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1552,30 +1560,45 @@ class KafkaApis(val requestChannel: RequestChannel,
         sendResponseMaybeThrottle(request, requestThrottleMs =>
           new AddPartitionsToTxnResponse(requestThrottleMs, partitionErrors.asJava))
       } else {
-        // Send response callback
-        def sendResponseCallback(error: Errors): Unit = {
-          def createResponse(requestThrottleMs: Int): AbstractResponse = {
-            val responseBody: AddPartitionsToTxnResponse = new AddPartitionsToTxnResponse(requestThrottleMs,
-              partitionsToAdd.asScala.map{tp => (tp, error)}.toMap.asJava)
-            trace(s"Completed $transactionalId's AddPartitionsToTxnRequest with partitions $partitionsToAdd: errors: $error from client ${request.header.clientId}")
-            responseBody
+        val (_, incorrectMessageFormat) = partitionsToAdd.asScala.partition { tp =>
+          replicaManager.getMagic(tp) match {
```

Comment:
The config lives in ZK. We have a listener that keeps the LogConfig in each Log up to date. The ReplicaManager  gets the config from the Log, which only works if the broker is hosting that partition.

The simple solution is to fetch from ZK, but it's not ideal to do it for every request. An alternative is to keep a cache for every topic. We can probably reuse the existing TopicConfigHandler to keep the cache updated.